activerecord/lib/active_record/connection_adapters/mysql_adapter.rb

```diff
@@ -417,7 +417,7 @@ def exec_stmt(sql, name, binds, cache_stmt: false)
           affected_rows = stmt.affected_rows
 
           stmt.free_result
-          stmt.close if binds.empty?
+          stmt.close if !cache_stmt
```

Comment:
This is follow up to https://github.com/rails/rails/commit/cbcdecd2c55fca9613722779231de2d8dd67ad02.
We should add `!cache_stmt` to this line.